source/MaterialXView/Material.cpp

```diff
@@ -224,8 +255,7 @@ bool Material::generateShader(mx::GenContext& context)
         return false;
     }
 
-    // TODO: Add support for transparency detection
-    _hasTransparency = false;
+    _hasTransparency = context.getOptions().hwTransparency;
```

Comment:
Where is the transparency option set?

It needs to be detected from the renderable MaterialX element. Or maybe set in the UI options if we want an easy way out for now.

Reply:
Yes. This is done just before we call generate() to create the shader. We just cache the value here since the context will be overwritten for the next shader generation.